src/pypromice/L0toL1.py

```diff
@@ -32,7 +33,8 @@ def toL1(L0, vars_df, flag_file=None, T_0=273.15, tilt_threshold=-100):
     assert(type(L0) == xr.Dataset)
     ds = L0
 
-    ds = _flagNAN(ds, flag_file)                                               # Flag NaNs
+    ds = _flagNAN(ds)                                                          # Flag NaNs
+    ds = _adjustData(ds)                                                       # Flag NaNs
```

Comment:
I agree that the flagging should occur here, but should data also be adjusted at the beginning of the `L0toL1` step?

I am wondering if adjusted data be accidentally modified in the subsequent steps (e.g. variable corrections and smoothing). As users are defining these adjustments on `L3` data, should adjusting instead be conducted after all variable corrections have been applied?

Reply:
From my experience, most of the adjustment should be done here.
So if there is a temporary offset or miscalibration, then it should be corrected before these values go to further filtering and into the correction of other variables.

In the GC-Net historical work, I am running the adjustment script a second time later on (in the equivalent of L2toL3), so that I can filter L3 variables for specific periods/sites.

Reply:
Good to know. I'll leave it in this place currently. Maybe we will adopt the same approach as how you are doing it for the historical work at a later date.

Reply:
Hey Penny,
I need to modify something there.
Currently the time shift (like the one needed at KPC_Uv3) is applied in adjustData. 
I'd like to specify the data to be flagged in true time period, not in "time period in shifted L0 file".
Right now the implementation does not allow that.

I'd also like to wrap this part (timeShift + flagData + adjustData) into a fail-safe "try" (at least for now). Because we don't want an improperly filled flag csv file to crash the whole code. Then we can later on go into details of the possible fails (wrong variable name, wrong date format, wrong adjustment function...etc.).

Reply:
That sounds good. Please feel free to change accordingly and push the new modifications.

Reply:
all of these points have been resolved:
- putting _adjustTime first
- wrap timeShift + flagData + adjustData into a fail-safe "try"